src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -150,6 +154,21 @@ public IActionResult Restart()
             return Ok(_webHostSettings);
         }
 
+        [HttpPost]
+        [Route("admin/host/offline")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public IActionResult Offline(bool isOffline)
+        {
+            if (_settingsManager.FileSystemIsReadOnly)
+            {
+                return BadRequest();
+            }
+
+            _scriptHostManager.SetOfflineState(isOffline);
```

Comment:
Could we avoid adding new APIs to `ScriptHostManager`? That type is being removed, and having more logic added there will make pulling these changes into `hosting` more difficult.